pkg/plugin/backup_item_action.go

```diff
@@ -38,41 +39,43 @@ import (
 // interface.
 type BackupItemActionPlugin struct {
 	plugin.NetRPCUnsupportedPlugin
-	impl arkbackup.ItemAction
-	log  *logrusAdapter
+
+	*serverMux
 }
 
 // NewBackupItemActionPlugin constructs a BackupItemActionPlugin.
-func NewBackupItemActionPlugin(itemAction arkbackup.ItemAction) *BackupItemActionPlugin {
+func NewBackupItemActionPlugin() *BackupItemActionPlugin {
 	return &BackupItemActionPlugin{
-		impl: itemAction,
+		serverMux: newServerMux(),
 	}
 }
 
-func (p *BackupItemActionPlugin) Kind() PluginKind {
-	return PluginKindBackupItemAction
-}
+//////////////////////////////////////////////////////////////////////////////
+// client code
+//////////////////////////////////////////////////////////////////////////////
 
-// GRPCServer registers a BackupItemAction gRPC server.
-func (p *BackupItemActionPlugin) GRPCServer(s *grpc.Server) error {
-	proto.RegisterBackupItemActionServer(s, &BackupItemActionGRPCServer{impl: p.impl})
-	return nil
-}
-
-// GRPCClient returns a BackupItemAction gRPC client.
+// GRPCClient returns a clientMux for BackupItemAction gRPC clients.
 func (p *BackupItemActionPlugin) GRPCClient(c *grpc.ClientConn) (interface{}, error) {
```

Comment:
The naming that we're locked into here (`GRPCClient(...)`) due to go-plugin is unfortunate, since what we're returning is not actually a gRPC client but something that makes clients. Might be helpful to expand the godoc to explain that this is called by go-plugin, and can return any type but it should be a client or something that can be used to create a client